packages/gcc10.rb

```diff
@@ -35,19 +39,27 @@ class Gcc10 < Package
   depends_on 'isl'
   depends_on 'glibc'
   depends_on 'zstd'
-
+  
   def self.preinstall
-    gccver = `gcc -v 2>&1 | tail -1 | cut -d' ' -f3`.chomp
-    abort "GCC version #{gccver} already installed.".lightgreen unless "#{gccver}" == "No" || "#{gccver}" == "not" || "#{gccver}" == "gcc:" || "#{gccver}" == "#{version}"
+    installed_gccver = `gcc -v 2>&1 | tail -1 | cut -d' ' -f3`.chomp
+    gcc_version = version.split("-")[0]
+    abort "GCC version #{installed_gccver} already installed.".lightgreen unless "#{installed_gccver}" == "No" || "#{installed_gccver}" == "not" || "#{installed_gccver}" == "gcc:" || "#{installed_gccver}" == "#{gcc_version}"
   end
 
   def self.build
+    # Set ccache sloppiness as per
+    # https://wiki.archlinux.org/index.php/Ccache#Sloppiness
+    system "ccache --set-config=sloppiness=file_macro,locale,time_macros"
+    # Prefix ccache to path.
+    ENV['PATH'] = "#{CREW_LIB_PREFIX}/ccache/bin:#{CREW_PREFIX}/bin:/usr/bin:/bin"
```

Comment:
What is wrong with putting ccache in a conventional bin directory?  We wouldn't need all this extra `PATH` manipulation then, correct?

Reply:
#{CREW_LIB_PREFIX}/ccache/bin just contains the special ccache symlinks for g++, gcc, etc which all point to /usr/local/bin/ccache or whatever.

This way when a program calls gcc, with the prepended path it actually calls the ccache wrapper.

(And this is ONLY for building gcc.) You don't need ccache installed otherwise.)

Reply:
Makes sense.  Carry on then...